app/src/main/java/org/gdg/frisbee/android/adapter/ContributorAdapter.java

```diff
@@ -1,60 +1,80 @@
+
 package org.gdg.frisbee.android.adapter;
 
 import android.content.Context;
+import android.os.Build;
 import android.text.TextUtils;
-import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
 import android.widget.ArrayAdapter;
 import android.widget.ImageView;
 import android.widget.TextView;
 
+import com.squareup.picasso.Picasso;
+
 import java.util.Collection;
 
 import org.gdg.frisbee.android.R;
 import org.gdg.frisbee.android.api.model.Contributor;
-import org.gdg.frisbee.android.app.App;
-
-/**
- * Created with IntelliJ IDEA.
- * User: maui
- * Date: 08.07.13
- * Time: 02:28
- * To change this template use File | Settings | File Templates.
- */
-public class ContributorAdapter extends ArrayAdapter<Contributor> {
 
-    private Context mContext;
-    private LayoutInflater mInflater;
+public class ContributorAdapter extends ArrayAdapter<Contributor> {
 
-    public ContributorAdapter(Context context, int textViewResourceId) {
-        super(context, textViewResourceId);
-        mContext = context;
-        mInflater = LayoutInflater.from(mContext);
+    public ContributorAdapter(Context context) {
+        super(context, R.layout.contributor_item);
     }
 
     @Override
-    public View getView(int position, View view, ViewGroup parent) {
-        if(view == null)
-            view = mInflater.inflate(R.layout.contributor_item, null);
+    public View getView(int position, View convertView, ViewGroup parent) {
+        if (convertView == null) {
+            convertView = View.inflate(getContext(), R.layout.contributor_item, null);
+
+            ViewHolder viewHolder = new ViewHolder();
+            viewHolder.contributorName = (TextView) convertView.findViewById(R.id.contributorName);
+            viewHolder.contributorAvatar = (ImageView) convertView.findViewById(R.id.contributorIcon);
+
+            convertView.setTag(viewHolder);
+        }
 
         Contributor contributor = getItem(position);
 
-        TextView name = (TextView)view.findViewById(R.id.contributorName);
-        name.setText(contributor.getLogin());
+        ViewHolder viewHolder = (ViewHolder) convertView.getTag();
+        viewHolder.contributorName.setText(contributor.getLogin());
+
         if (!TextUtils.isEmpty(contributor.getAvatarUrl())) {
-            final ImageView c = (ImageView) view.findViewById(R.id.contributorIcon);
-            App.getInstance().getPicasso()
-                    .load(contributor.getAvatarUrl())
-                    .into(c);
+            Picasso.with(getContext()).load(contributor.getAvatarUrl()).into(viewHolder.contributorAvatar);
         }
-        return view;
+
+        return convertView;
     }
 
     @Override
     public void addAll(Collection<? extends Contributor> contributors) {
-        for (Contributor c : contributors) {
-            add(c);
+        if (Build.VERSION.SDK_INT > Build.VERSION_CODES.GINGERBREAD_MR1) {
+            super.addAll(contributors);
+        } else {
```

Comment:
Need to add `setNotifyOnChange(false)` and then `setNotifyOnChange(true) after all items are added. And then notify it manually.

See implementation:
```
    /**
     * Adds the specified Collection at the end of the array.
     *
     * @param collection The Collection to add at the end of the array.
     */
    public void addAll(Collection<? extends T> collection) {
        synchronized (mLock) {
            if (mOriginalValues != null) {
                mOriginalValues.addAll(collection);
            } else {
                mObjects.addAll(collection);
            }
        }
        if (mNotifyOnChange) notifyDataSetChanged();
    }
```

and

```
    /**
     * Adds the specified object at the end of the array.
     *
     * @param object The object to add at the end of the array.
     */
    public void add(T object) {
        synchronized (mLock) {
            if (mOriginalValues != null) {
                mOriginalValues.add(object);
            } else {
                mObjects.add(object);
            }
        }
        if (mNotifyOnChange) notifyDataSetChanged();
    }
```